src/core_landice/mode_forward/mpas_li_calving.F

```diff
@@ -1249,6 +1256,794 @@ subroutine eigencalving(domain, err)
+          ! Compute the hydrostatic to tensile stress ratio
+
+         !do iCell = 1, nCells
+         !    if (li_mask_is_floating_ice(cellMask(iCell))) then
+         !        floatingBasalMassBal(iCell) = 0.2/31536000.0*tanh((lowerSurface(iCell)-bedTopography(iCell))/75.0) * max(-100-lowerSurface(iCell),0.0)
+         !    end if
+         !end do
+         ! Temporary change of BMB for the MISMIP+ Ice1r experiment
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             source(iCell) = 0.0_RKIND
+         else
+             source(iCell) = nstar(iCell) * (1.0d0-szero(iCell))*epsmax(iCell)-floatingBasalMassBal(iCell)/rhoi/(thickness(iCell))
+         endif
+         enddo
+
+         seconds = daysSinceStart*24.0*3600.0
+
+         ! below is the force for the manufacured experiments
+         !force(:) = -0.44/2.0 * (uReconstructX(1,:)/1000.0*(1+epsmax(:)*seconds)*exp(-uReconstructX(1,:)*seconds/1000.0) +    &
+         !    source(:) * (1+exp(-uReconstructX(1,:)*seconds/1000.0)))
+         !force(:) = -0.44/2.0 * (uReconstructX(1,:)/1000.0*exp(-uReconstructX(1,:)*seconds/1000.0) +    &
+         !    source(:) * (1+exp(-uReconstructX(1,:)*seconds/1000.0)))
+
+         ! set force to zero if not manufactured
+         force(:) = 0.0
+
+         ddamagedt(:) = source(:) * damage(:) + force(:)
+
+         damage(:) = damage(:) + ddamagedt(:) * deltat
+
+         !damage(:) =  0.5
+         ! temporarily set damage to a constant value
+
+         do iCell = 1, nCells
+         if (thickness(iCell) == 0.0_RKIND) then
+             damage_nye(iCell) = 0.0_RKIND
+         else
+             damage_nye(iCell) = (2.0+alpha(iCell)) * sigmamax(iCell) / (gravity*(rhoo-rhoi)*thickness(iCell))
+         endif
+         enddo
+         !damage_nye(:) = 0.5
+         ! temporarily set damage_nye to a constant value, 0.8
+
+         do iCell = 1, nCells
+            if (damage(iCell) > config_damage_restore_threshold) then
+                damage_max(iCell) = damage(iCell)
+            end if
+         end do
+         ! save the damage_max value for restoring (no heal) later if we want
+
+         do iCell = 1, nCells
+            if (damage(iCell) < 0.0_RKIND) then
+                damage(iCell) = 0.0_RKIND
+            end if
+            if (damage(iCell) < damage_nye(iCell)) then
+                damage(iCell) = damage_nye(iCell)
+            end if
+            if (damage(iCell) > 1.0_RKIND) then
+                damage(iCell) = 1.0_RKIND
+            end if
+         end do
+
+
+         !do iCell = 1, nCells
+         !   if ((li_mask_is_grounded_ice(cellMask(iCell))) .or. (.not.(li_mask_is_dynamic_ice(cellMask(iCell))))) then
+         !       damage(iCell) = 0.0_RKIND
+         !   end if
+         !end do
+
+         do iCell = 1, nCells
+            if ((li_mask_is_grounded_ice(cellMask(iCell))) .or. (thickness(iCell) .eq. 0.0_RKIND)) then
+                damage(iCell) = 0.0_RKIND
```

Comment:
So we do not calculate damage on grounded ice?

Reply:
No, we don't

Reply:
We've talked w/ Jeremy about this before too. I think there is no reason why we can't calc. damage for grounded ice but I think their theory was specifically developed for floating ice. So we might need to discuss a bit more if we also want to calc. damage on grounded ice.

Reply:
@tongzhangice , let's add a comment here explaining that the theory we are using it meant to only apply to grounded ice.  Also, can you describe this in the description of the subroutine above the subroutine definition?

Reply:
Comment and description are added.